app/cdap/services/datasource/DataSourceHttp.ts

```diff
@@ -0,0 +1,460 @@
+      rx: subject,
+      resource: generatedResource,
+      type: 'POLL',
+      excludeFromHealthCheck,
+    };
+    this.polling[generatedResource.id] = bindingInfo;
+
+    debugLog(`Polling requested for ${generatedResource.id} URL ${generatedResource.url}`);
+
+    const responseHandler = this.createResponseHandler(bindingInfo);
+    Observable.ajax(generatedResource).subscribe(responseHandler);
+
+    return observable;
+  }
+
+  public startClientPoll = (resourceId) => {
+    const interval = objectQuery(this.polling, resourceId, 'resource', 'intervalTime');
+    debugLog(`Setting timeout for ${resourceId} with interval ${interval}`);
+    const intervalTimer = setTimeout(() => {
+      const bindingInfo = this.polling[resourceId];
+      const resource = objectQuery(bindingInfo, 'resource');
+      if (!resource || !WindowManager.isWindowActive()) {
+        clearTimeout(intervalTimer);
+        return;
+      }
+      const responseHandler = this.createResponseHandler(bindingInfo);
+      Observable.ajax(resource).subscribe(responseHandler);
+    }, interval);
+    return intervalTimer;
+  };
+
+  public stopPoll(resourceId) {
+    let id;
+    if (typeof resourceId === 'object' && resourceId !== null) {
+      id = resourceId.params.pollId;
+    } else {
+      id = resourceId;
+    }
+
+    if (this.polling[id]) {
+      debugLog(`Stopping polling for ${id}`);
+      clearTimeout(this.polling[id].resource.interval);
+      this.polling[id].rx.complete();
+      delete this.polling[id];
+    }
+  }
+
+  public pausePoll = () => {
+    Object.keys(this.polling)
+      .filter((subscriptionID) => this.polling[subscriptionID].type === 'POLL')
+      .forEach((subscriptionID) => {
+        clearTimeout(this.polling[subscriptionID].resource.interval);
+        this.polling[subscriptionID].resource.interval = null;
+      });
+  };
+
+  public resumePoll = () => {
+    // TODO When resuming, should polling requests be staggered to avoid overloading the server?
+    Object.keys(this.polling)
+      .filter((subscriptionID) => this.polling[subscriptionID].type === 'POLL')
+      .forEach((subscriptionID) => {
+        this.polling[subscriptionID].resource.interval = this.startClientPoll(subscriptionID);
+      });
+  };
+
+  public destroy() {
+    // stopping existing polls
+    for (const key in this.polling) {
+      if (this.polling[key].type === 'POLL') {
+        this.stopPoll(this.polling[key].resource.id);
+      }
+    }
+    this.polling = {};
+  }
+
+  public buildUrl(url: string, params: IParamsMap = {}) {
+    if (!params) {
+      return url;
+    }
+    const parts = [];
+
+    const forEachSorted = (obj: IParamsMap, iterator) => {
+      const keys = Object.keys(params).sort();
+      keys.forEach((key) => {
+        iterator.call(null, obj[key], key);
+      });
+      return keys;
+    };
+
+    const encodeUriQuery = (val: string, pctEncodeSpaces?: boolean) => {
+      return encodeURIComponent(val)
```

Comment:
is there a better way to do this?

Reply:
Probably. I filed https://cdap.atlassian.net/browse/CDAP-19018 to track but I won't tackle in this PR.